packages/sheets-ui/src/views/formula-bar/FormulaBar.tsx

```diff
@@ -61,34 +61,46 @@ export function FormulaBar() {
     }
 
     useLayoutEffect(() => {
-        const workbook = univerInstanceService.getCurrentUnitForType<Workbook>(UniverInstanceType.UNIVER_SHEET)!;
-        const subscription = merge(
-            worksheetProtectionRuleModel.ruleChange$,
-            rangeProtectionRuleModel.ruleChange$,
-            selectionManager.selectionMoveEnd$,
-            workbook.activeSheet$
-        ).pipe(
-            switchMap(() => {
-                const unitId = workbook.getUnitId();
-                const worksheet = workbook.getActiveSheet();
-                if (!worksheet) return EMPTY;
+        const subscription = univerInstanceService.getCurrentTypeOfUnit$<Workbook>(UniverInstanceType.UNIVER_SHEET).pipe(
+            switchMap((workbook) => {
+                if (!workbook) {
+                    return EMPTY;
+                }
+                return workbook.activeSheet$.pipe(
+                    switchMap((worksheet) => {
+                        if (!worksheet) {
+                            return EMPTY;
+                        }
+                        return merge(
+                            worksheetProtectionRuleModel.ruleChange$,
+                            rangeProtectionRuleModel.ruleChange$,
+                            selectionManager.selectionMoveEnd$
+                        ).pipe(
+                            switchMap(() => {
+                                const unitId = workbook.getUnitId();
+                                const worksheet = workbook.getActiveSheet();
```

Comment:
You use `worksheet` in line 70. It is not necessary to call `getActiveSheet` here.